### PR TITLE
Rendered layer library from API

### PR DESCRIPTION
**Pull request by andresgnlez**

### CODE REVIEW

File: app/src/actions/layers.js [hunk 1]
```diff
@@ -7,9 +7,39 @@ import {
...
+      // moves "args" content to the root of the object
+      Object.assign(l, l.args);
+      // removes "args" property from the object
+      /* eslint no-param-reassign: 0 */
+      delete l.args;
+    });
+
+
+    // Match workspace ids with library ones
+    const matchedLayers = _.filter(libraryLayers, (l) => workspaceLayersIds.indexOf(l.id) !== -1);
```
Comment (nerik) [inline app/src/actions/layers.js]: One letter variable. Also, remove unneeded `()`

File: app/src/actions/layers.js [hunk 2]
```diff
@@ -7,9 +7,39 @@ import {
...
+
+    matchedLayers.forEach((l) => {
+      const localLayer = _.find(workspaceLayers, (wl) => wl.id === l.id);
```
Comment (nerik) [inline app/src/actions/layers.js]: One letter variable. Also, remove unneeded `()`

File: app/src/components/Map.jsx [hunk 3]
```diff
@@ -295,6 +296,7 @@ Map.propTypes = {
   token: React.PropTypes.string,
   tilesetUrl: React.PropTypes.string,
   setZoom: React.PropTypes.func,
+  getLayerLibrary: React.PropTypes.func,
   getWorkspace: React.PropTypes.func,
```
Comment (nerik) [inline app/src/components/Map.jsx]: Remove `getWorkspace` from propTypes

File: app/src/containers/Map.js [hunk 4]
```diff
@@ -30,6 +33,9 @@ const mapStateToProps = (state) => ({
 });
 
 const mapDispatchToProps = (dispatch, ownProps) => ({
+  getLayerLibrary: () => {
+    dispatch(getLayerLibrary());
+  },
   getWorkspace: () => {
```
Comment (nerik) [inline app/src/containers/Map.js]: Remove getWorkspace from mapDispatchToProps + import

File: app/src/containers/LayerLibrary.js [hunk 5]
```diff
@@ -1,16 +1,24 @@
...
   closeModal: () => {
     dispatch(setLayerLibraryModalVisibility(false));
   },
+  addLayer: (layerId) => {
+    // TODO add layer logic
+    console.warn(layerId, 'adds layer');
+  },
+  removeLayer: (layerId) => {
+    // TODO remove layer logic
+    console.warn(layerId, 'removes layer');
```
Comment (nerik) [inline app/src/containers/LayerLibrary.js]: layerId is undefined

Comment (andresgnlez) [inline app/src/containers/LayerLibrary.js]: It's WIP. It's next part gonna implement.

**Comment by tiagojsag** [comment]
Looks good so far, but I would not merge until the vessels layer is back, otherwise we might block a lot of things.

**Pull request closed without merging**